app/javascript/packages/document-capture/components/warning.jsx

```diff
@@ -13,7 +15,7 @@ import PageHeading from './page-heading';
  * @prop {(() => void)=} actionOnClick Primary action button text.
  * @prop {import('react').ReactNode} children Component children.
  * @prop {string=} troubleshootingHeading Heading text preceding troubleshooting options.
- * @prop {(TroubleshootingOption[])=} troubleshootingOptions Array of troubleshooting options.
+ * @prop {TroubleshootingOption[]|ReactNode=} troubleshootingOptions Troubleshooting options.
```

Comment:
Alternatively, instead of overloading this usage to support either the element or the array of options, we could enforce the element be passed always. A difference is that we default a heading in the case that someone passes `troubleshootingOptions` but not `troubleshootingHeading`, but maybe that default could be moved to `TroubleshootingOptions` itself.

Reply:
So you're saying troubleshooting options would look like this?
```
{
  heading: 'aaa',
  options: [ { text, url  },  { text, url } ],
}
```
That makes sense to me, but the current overload is not so complex either

Reply:
I was imagining to remove support for `troubleshootingHeading` and `troubleshootingOptions` as an array, and only to only support it passed as an element.

Reply:
Oh so more like a `children` passthrough?

Reply:
> Oh so more like a `children` passthrough?

Yeah, although `Warning` already accepts children for content of the warning, so we'd need it to be a different prop. I figure we can use the existing `troubleshootingOptions` for this.

So usage ends up being like:

```jsx
<Warning
  troubleshootingOptions={<TroubleshootingOptions {/*...*/} />}
>
  Warning content
</Warning>
```

Reply:
I'll plan to make this refactoring, but noting a couple watch items for the future:

- If we're changing pairing props "troubleshootingHeading" and "troubleshootingOptions" to a single "options" prop, it raises a question of what to do about "actionText" and "actionOnClick", as they are similar in nature.
   - It's not as obvious to allow an element to be passed through for the action, since we want to enforce an opinionated appearance of the button ("big", "primary").
- Noting that we have a similar ERB file for this markup, and options ideally align ([`idv/shared/_error.html.erb`](https://github.com/18F/identity-idp/blob/d3de31e64e890f358a221f075f5e5f2f6e59024c/app/views/idv/shared/_error.html.erb))
   - Proposed refactoring arguably brings into closer alignment, since "options" is a singular setting there, and "heading" has smart defaults.